belgif-rest-problem-spring/src/main/java/io/github/belgif/rest/problem/spring/InvalidRequestExceptionHandler.java

```diff
@@ -0,0 +1,217 @@
+        if (original != null) {
+            String actual = message.getContext()
+                    .flatMap(ValidationReport.MessageContext::getRequestPath)
+                    .orElse("");
+
+            String regex = original.replace("{" + name + "}", "(.*)").replaceAll("/\\{.*}", "/.*");
+
+            Pattern pattern = Pattern.compile(regex);
+            Matcher matcher = pattern.matcher(actual);
+
+            if (matcher.find()) {
+                return matcher.group(1);
+            }
+        }
+
+        return null;
+    }
+
+    private InputValidationIssue handleSpecialCases(ValidationReport.Message message) {
+        switch (message.getKey()) {
+            case "validation.schema.invalidJson":
+            case "validation.request.body.schema.invalidJson": {
+                return InputValidationIssues.schemaViolation(InEnum.BODY, null, null, "Unable to parse JSON");
+            }
+            case "validation.schema.unknownError": {
+                LOGGER.error("An unknown error occured during schema validation: {}", message.getMessage());
+                return InputValidationIssues.schemaViolation(null, null, null,
+                        "An error occurred during schema validation");
+            }
+            default:
+                return null;
+        }
+    }
+
+    private InputValidationIssue buildIssue(ValidationReport.Message message, InEnum in, String name,
+            String value, String detail) {
+        switch (message.getKey()) {
+            case "validation.request.parameter.query.unexpected":
+                return InputValidationIssues.unknownInput(in, name, value);
+            default:
+                return InputValidationIssues.schemaViolation(in, name, value, detail);
+        }
+    }
+
+    private boolean isNonExistingPath(InvalidRequestException ex) {
+        return ex.getValidationReport().getMessages().stream()
+                .anyMatch(m -> "validation.request.path.missing".equals(m.getKey()));
+    }
+
+    private List<ValidationReport.Message> getExplodedMessages(InvalidRequestException ex) {
+        List<ValidationReport.Message> messages = new ArrayList<>();
+        for (ValidationReport.Message message : ex.getValidationReport().getMessages()) {
+            if ("validation.request.body.schema.allOf".equals(message.getKey()) && message.getNestedMessages() != null
+                    && !message.getNestedMessages().isEmpty()) {
+                messages.addAll(message.getNestedMessages());
+            } else {
+                messages.add(message);
+            }
+        }
+        return messages;
+    }
+
+    private JsonNode getRequestBody(HttpServletRequest request) {
+        try {
+            InputStream inputStream = request.getInputStream();
+            return mapper.readTree(inputStream);
+        } catch (IOException ex) {
+            if (!"GET".equals(request.getMethod()) && !"HEAD".equals(request.getMethod())) {
```

Comment:
I guess you get IOException when there is no body? See my earlier comment about only parsing the request body when necessary.

Reply:
Or when there is an actual error, I kept the error log, but removed the if block to verify the request method.